### ISSUE TITLE
RawPedia: Update MediaWiki to new LTS version

### ISSUE THREAD

**Issue by Thanatomanic** [Beep6581/RawTherapee#6272]
The version of MediaWiki we use for RawPedia (1.31.x) has reached its EOL: https://www.mediawiki.org/wiki/Version_lifecycle
Can we upgrade to the new 1.35.x LTS version?

I'm not 100% sure if it's worth it, but one thing I don't find particularly 'modern' for our current wiki is the editor. If there is a something we can do about that I would find that most welcome. There are now very limited options to help write a good page and we need a lengthy page explaining all the features... http://rawpedia.rawtherapee.com/Coding_Rawpedia_pages

**Comment by TechXavAL** [comment]
As you say, maybe it's not worth it.

The fact that the version RawPedia runs has reached its EOL shouldn't have any impact on this server. And there's the old mantra: if it works...

Mediawiki it's not famous for its content creation support. Nor is it for being "responsive friendly". And above all, it is not famous for its support to modern HTML features.

What that means is:
* the code editor is what it is, and they know it is "not good". You have a few alternatives, all relying on extra extensions, with their own share of problems. In other words: to my knowledge there's no better code editor in the latest version
* if you want something close to what Wikipedia has, you have to install a bunch of extensions and manage, customize and support them yourself (as you have hacked them)
* if you want to adhere to what Mediawiki offers to you, you just have to stick to the 3-4 pages of explanations they have published (for text and images). And you will be very, very limited.
* if you want extra features for RawPedia, the only solution for now is to follow that lengthy page, which lists a good quantity of things not possible in vanilla Mediawiki. Specially for the responsive tables.

If creating content has its own challenges, you have to sum up the inability of Mediawiki for doing certain things considered as a _de facto_ in other situations (like creating HTML pages from scratch).

However, if you just copy/paste the sample code of that lengthy page, and then customize it to suit the needs of your page, it is possible to do a lot of things. At the cost of extra markup in the code.

An example: try to create a Mediawiki gallery of 3 images, or a row of 3 images, which fills the available space (around 1/3 of the horizontal available space for each image), and then make it work nicely for smaller than FullHD screens... All kind of smaller screens... Good luck!

**Comment by Thanatomanic** [comment]
I understand what you're saying Xavier. If you look at https://www.mediawiki.org/wiki/Editor and reading about VisualEditor (https://www.mediawiki.org/wiki/Extension:VisualEditor) it seems that upgrading to 1.35 makes live easier to get this VisualEditor working as well. This seems to me like a big upgrade over our current small button bar.
This will still not make more advanced markup possible like your description of a responsive gallery though.

**Comment by TechXavAL** [comment]
Yes. For occasional and simple edits most probably you're right.

I must admit that I almost never use the button bar and instead type the code straight away (it's easy when you learn which characters you have to type).

Anyway, I think it's a decision for @Beep6581 to make.
